frontend/src/app/resource-detail/resource-detail.component.ts

```diff
@@ -10,38 +11,40 @@ import { ActivatedRoute } from '@angular/router';
 })
 export class ResourceDetailComponent implements OnInit {
   resource: Resource;
+  mapLoc: LatLngLiteral;
 
   constructor(
     private api: ApiService,
     private route: ActivatedRoute
   ) {
     this.route.params.subscribe(params => {
       const resourceId = params.resourceId ? parseInt(params.resourceId, 10) : null;
-      if (this.route.snapshot.data['title'] == 'Event Details') {
-        if (isFinite(resourceId)) {
-          this.api.getEvent(resourceId).subscribe(resource => {
-            this.resource = resource;
-          });
-        }
-      } else if (this.route.snapshot.data['title'] == 'Location Details') {
-        if (isFinite(resourceId)) {
-          this.api.getLocation(resourceId).subscribe(resource => {
-            this.resource = resource;
-          });
-        }
-      } else if (this.route.snapshot.data['title'] == 'Resource Details') {
-        if (isFinite(resourceId)) {
-          this.api.getResource(resourceId).subscribe(resource => {
-            this.resource = resource;
-          });
-        }
+
```

Comment:
Thank you for cleaning this up! I was really dissatisfied with how I wrote this but never circled back to make it better :stuck_out_tongue_closed_eyes: